lib/smart_proxy_dynflow/continuous_output.rb

```diff
@@ -37,18 +37,20 @@ def humanize
       raw_outputs.map { |output| output['output'] }.join("\n")
     end
 
-    def add_exception(context, exception, timestamp = Time.now.getlocal)
-      add_output(context + ": #{exception.class} - #{exception.message}", 'debug', timestamp)
+    def add_exception(context, exception, timestamp: Time.now.getlocal, id: nil)
```

Comment:
Isn't this an API change?
```ruby
def add_exception(context, exception, timestamp = 'now')
  puts timestamp
end

def add_exception_new(context, exception, timestamp: 'now')
  puts timestamp
end

add_exception('a', 'b')
add_exception('a', 'b', 'explicit')

add_exception_new('a', 'b')
add_exception_new('a', 'b', 'explicit')
```

```console
$ ruby test2.rb
now
explicit
now
test2.rb:5:in `add_exception_new': wrong number of arguments (given 3, expected 2) (ArgumentError)
	from test2.rb:13:in `<main>'
```

Or do you accept it since it only appears to be used internally.

Reply:
While this isn't strictly private api, none of the smart proxy plugins which depend on sp-dynflow seem to use this, so I'd be willing to let it slide.